eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/bag/MutableBagFactory.java

```diff
@@ -48,8 +49,43 @@
         return this.with(elements);
     }
 
+    default <T> MutableBag<T> ofOccurrences(T element, int occurrence)
+    {
+        return this.withOccurrences(element, occurrence);
+    }
+
+    default <T> MutableBag<T> ofOccurrences(T element1, int occurrence1, T element2, int occurrence2)
+    {
+        return this.withOccurrences(element1, occurrence1, element2, occurrence2);
+    }
+
+    default <T> MutableBag<T> ofOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3)
+    {
+        return this.withOccurrences(element1, occurrence1, element2, occurrence2, element3, occurrence3);
+    }
+
+    default <T> MutableBag<T> ofOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3, T element4, int occurrence4)
+    {
+        return this.withOccurrences(element1, occurrence1, element2, occurrence2, element3, occurrence3, element4, occurrence4);
+    }
+
+    default <T> MutableBag<T> ofOccurrences(ObjectIntPair<T>... elementsWithOccurrences)
+    {
+        return this.withOccurrences(elementsWithOccurrences);
+    }
+
     <T> MutableBag<T> with(T... elements);
 
+    <T> MutableBag<T> withOccurrences(T element, int occurrence);
```

Comment:
We have a choice to make here.  The specification and implementation of this method looks fine, but will require the next version of Eclipse Collections to be 11.0 because the `withOccurrences` methods do not have default implementations.  As long as @nikhilnanivadekar is ok with this, then I am fine.  I would suggest adding `@since 11.0` tags to the interfaces for `MutableBagFactory` and `ImmutableBagFactory` then as this will require a major release.